Dockerfile.prod

```diff
@@ -0,0 +1,29 @@
+# Creating multi-stage build for production
+FROM node:16-alpine as build
+RUN apk update && apk add --no-cache build-base gcc autoconf automake zlib-dev libpng-dev vips-dev > /dev/null 2>&1
+ARG NODE_ENV=production
+ENV NODE_ENV=${NODE_ENV}
+
+WORKDIR /opt/
+COPY package.json yarn.lock ./
+RUN yarn config set network-timeout 600000 -g && yarn install --production
+ENV PATH /opt/node_modules/.bin:$PATH
+WORKDIR /opt/app
+COPY . .
+RUN yarn build
+
+# Creating final production image
+FROM node:16-alpine
+RUN apk add --no-cache vips-dev
+ARG NODE_ENV=production
+ENV NODE_ENV=${NODE_ENV}
+WORKDIR /opt/
+COPY --from=build /opt/node_modules ./node_modules
+WORKDIR /opt/app
+COPY --from=build /opt/app ./
+ENV PATH /opt/node_modules/.bin:$PATH
+
+RUN chown -R node:node /opt/app
+USER node
+EXPOSE 1337
+CMD ["yarn", "start"]
```

Comment:
why do we need a dockerfile for prod? Why not using the same and deal with differences just using envs?

Reply:
Different optimisations to begin with. Splitting development vs production images allows you to have production images that are as bare bones as possible, without affecting development experience. It is also defined in this manner in strapi docs.

We could probably unify them, but I think it might end up being less readable in comparison.

Reply:
but the content is almost the same, do we want to separate? 
duplicating the code just makes it harder to read and maintain, not easier, right? 

There's way to optimise it for prod without having 2 different images, and makes sense to have dev to be as close as possible to prod. For development we will probably not even use docker and just launch the yarn app

Reply:
Sure, I can get rid of it. I've just followed the documentation on this, no hard preference.